app/models/principal.rb

```diff
@@ -66,7 +68,7 @@ def associate_firm
   end
 
   def match_fca_number
-    unless Lookup::Firm.exists?(fca_number: self.fca_number)
+    unless self.fca_number.blank? || Lookup::Firm.exists?(fca_number: self.fca_number)
```

Comment:
On the `validate :match_fca_number` you can add an `if: ->{ fca_number.present? }` and take it out of this extra conditional here. Bit cleaner, WDYT?